File: Leetcode/Java/125. Valid Palindrome/solution.java
// https://leetcode.com/problems/valid-palindrome/

class Solution {
    public boolean isPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;

        while (left < right){
            // Skip not char or number index
            while (left < right && !Character.isLetterOrDigit(s.charAt(left))){
                left += 1;
            }
            while (left < right && !Character.isLetterOrDigit(s.charAt(right))){
                right -= 1;
            }

            if (Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right))){
                return false;
            }
            left += 1;
            right -= 1;     
        }

        return true;
    }
}

File: Leetcode/Typescript/125. Valid Palindrome/solution.ts
// https://leetcode.com/problems/valid-palindrome/description/

function isPalindrome(s: string): boolean {
  const isAlnum = /[0-9a-zA-Z]/; // For check alphabet, number
  let left = 0,
    right = s.length - 1;

  while (left < right) {
    // Skip space
    while (left < right && !isAlnum.test(s[left])) {
      left += 1;
    }
    while (left < right && !isAlnum.test(s[right])) {
      right -= 1;
    }

    if (s[left].toLowerCase() != s[right].toLowerCase()) {
      return false;
    }
    left += 1;
    right -= 1;
  }

  return true;
}
